refactoring: [Net Processing] Follow-ups to #21160

jnewbery opened the pull request:
#21160 ([net/net processing]: Move tx inventory into net_processing) had some unaddressed review comments when it was merged. This branch addresses those comments.

jnewbery committed:
net_processing: move CNode data access out of lock

CNode::m_relays_tx and CNode::m_bloom_filter_loaded access don't require
the Peer::TxRelay::m_bloom_filter_mutex lock, so move them out of the
lock scope.

See https://github.com/bitcoin/bitcoin/pull/21160#discussion_r736785417
and https://github.com/bitcoin/bitcoin/pull/21160#discussion_r736785662.

jnewbery commented:
@theuni I believe this addresses all your remaining review comments from #22160

fanquake commented:
cc @dergoegge

dergoegge commented on src/net_processing.cpp:
```diff
@@ -1236,7 +1236,7 @@ void PeerManagerImpl::InitializeNode(CNode *pnode)
     }
     if (!pnode->IsInboundConn()) {
-        PushNodeVersion(*pnode, *peer);
+        PushNodeVersion(*pnode, peer->m_tx_relay != nullptr);
```
```suggestion
        PushNodeVersion(*pnode, peer->m_tx_relay == nullptr);
```

jnewbery replied:
Fixed. Thanks!

dergoegge commented on src/net_processing.cpp:
```diff
@@ -2675,7 +2675,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // We send our version message in response.
         if (pfrom.IsInboundConn()) {
-            PushNodeVersion(pfrom, *peer);
+            PushNodeVersion(pfrom, peer->m_tx_relay != nullptr);
```
```suggestion
            PushNodeVersion(pfrom, peer->m_tx_relay == nullptr);
```

jnewbery replied:
Fixed. Thanks!

dergoegge reviewed (changes requested):
The tests are failing because the wrong value is being passed to `PushNodeVersion` for the `block_relay_only` param.

jnewbery commented:
> The tests are failing because the wrong value is being passed to PushNodeVersion for the block_relay_only param.

Oops. Now fixed. Thank you!

ajtowns commented on src/net_processing.cpp:
```diff
@@ -2675,7 +2675,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // We send our version message in response.
         if (pfrom.IsInboundConn()) {
-            PushNodeVersion(pfrom, *peer);
+            PushNodeVersion(pfrom, peer->m_tx_relay == nullptr);
```
Duplicating `peer->m_tx_relay == nullptr` at both call sites rather than just leaving it in the function seems kind of backwards? Also seems demonstrably a bit bug-prone, presumably warranting a `/*block_relay_only=*/` at least.

MarcoFalke replied:
Alternatively, pass the tx_relay pointer?

jnewbery replied:
Reverted change so this passes in a `const Peer&`

MarcoFalke commented:
See also suggestion in https://github.com/bitcoin/bitcoin/pull/21160#discussion_r837238573

jnewbery committed:
[fuzz] Assert that Peer.m_tx_relay.m_relay_txs has been set correctly

jnewbery committed:
[net processing] PushNodeVersion() takes a const Peer&

The peer object is not mutated by PushNodeVersion, so pass a const reference

jnewbery commented:
> See also suggestion in https://github.com/bitcoin/bitcoin/pull/21160#discussion_r837238573

Added.

I've removed the _net_processing: MaybeSendFeeFilter() takes a Peer::TxFilter*_ and _net_processing: PushNodeVersion() takes a bool instead of a Peer&_ commits. There doesn't seem to be agreement on what should be passed down to these functions, and in the absence of agreement, continuing to pass just a reference to the `Peer` object seems fine. I have however changed the function signature of `PushNodeVersion()` to take a const reference to Peer. That seems like an uncontroversial improvement.

MarcoFalke commented:
I think you forgot to force push your local branch?

MarcoFalke commented:
review ACK a40978dcbd6608695bc7f5191c4d0a3e48cbca0b

dergoegge commented:
ACK a40978dcbd6608695bc7f5191c4d0a3e48cbca0b

ajtowns commented:
ACK a40978dcbd6608695bc7f5191c4d0a3e48cbca0b
